### PR TITLE
[HOPSWORKS-1425], [HOPS-1518] and make it possible to set a different version of ndb for hops-metadata

### PR DESCRIPTION
**Pull request by berthoug**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-1425

### CODE REVIEW

File: templates/default/core-site.xml.erb [hunk 1]
```diff
@@ -201,6 +201,11 @@
     <value><%= @defaultFS %></value>
   </property>
 
+    <property>
+    <name>fs.defaultFS.alternativeScheme</name>
+    <value><%= @alternativeScheme %></value>
```
Comment (SirOibaf) [inline templates/default/core-site.xml.erb]: The `@alternativeScheme` works only if you pass it as a variable in the template resource in the default.rb recipe - I think in this case we can just hard code `hdfs` here.

File: templates/default/core-site.xml.erb [hunk 2]
```diff
@@ -201,6 +201,11 @@
     <value><%= @defaultFS %></value>
   </property>
 
+    <property>
```
Comment (SirOibaf) [inline templates/default/core-site.xml.erb]: Indentation

File: recipes/ndb.rb [hunk 3]
```diff
@@ -4,6 +4,10 @@
 ndb_connectstring()
 my_ip = my_private_ip()
 
+if ['hops']['ndb']['version']!=""
+  node['ndb']['version']=['hops']['ndb']['version']
```
Comment (SirOibaf) [inline recipes/ndb.rb]:
```suggestion
node.override['ndb']['version'] = node['hops']['ndb']['version']
```

File: recipes/ndb.rb [hunk 4]
```diff
@@ -4,6 +4,10 @@
 ndb_connectstring()
 my_ip = my_private_ip()
 
+if ['hops']['ndb']['version']!=""
```
Comment (SirOibaf) [inline recipes/ndb.rb]:
```suggestion
if node['hops']['ndb']['version'] != ""
```

File: recipes/default.rb [hunk 5]
```diff
@@ -51,7 +51,8 @@
   nn_https_address = private_recipe_ip("hops", "nn") + ":#{node['hops']['nn']['https_port']}"
 end
 
-defaultFS = "hdfs://#{nn_rpc_address}"
+defaultFS = "hopsfs://#{nn_rpc_address}"
+alternativeScheme = "hdfs"
```
Comment (SirOibaf) [inline recipes/default.rb]: See comment below

File: templates/default/core-site.xml.erb (hunk 2 above)
Comment (berthoug) [reply templates/default/core-site.xml.erb]: fixed

File: templates/default/core-site.xml.erb (hunk 1 above)
Comment (berthoug) [reply templates/default/core-site.xml.erb]: Yea, I missed the default.rb part. I guess we can hardcode for now and change in the future if it become needed.

File: recipes/default.rb (hunk 5 above)
Comment (berthoug) [reply recipes/default.rb]: fixed

File: templates/default/core-site.xml.erb (hunk 1 above)
Comment (berthoug) [reply templates/default/core-site.xml.erb]: especially as it is hardcoded in the default.rb :D

File: recipes/ndb.rb (hunk 3 above)
Comment (berthoug) [reply recipes/ndb.rb]: thanks

File: recipes/ndb.rb (hunk 4 above)
Comment (berthoug) [reply recipes/ndb.rb]: fixed

**Pull request merged**